### ISSUE TITLE
Users should not be allowed to place [Out] on `in` parameters

### ISSUE THREAD

**Issue by OmarTawfik** [dotnet/roslyn#22640]
```csharp
public class C {
    public void M([Out] in int x) {
    }
}
```
**Expected:** a compiler error.
**Actual:** attribute is emitted.

cc @jcouv @tannergooding

**Comment by OmarTawfik** [comment]
I'll take this one.
